compiler/plc_diagnostics/src/diagnostics/error_codes/E121.md

```diff
@@ -0,0 +1,29 @@
+# E121: Recursive type alias
+
+This error occurs when type aliases reference each other in a cycle, creating an infinite recursion.
+
+## Example
+
+```st
+TYPE type1 : type2; END_TYPE
+TYPE type2 : type1; END_TYPE
+```
+
+In this example, `type1` is defined as `type2`, which is in turn defined as `type1`, creating a circular dependency.
+
+## Another example
+
+```st
+TYPE self_type : self_type; END_TYPE
+```
+
+This shows a type alias that directly references itself.
+
+## How to fix
+
+Break the circular dependency by ensuring that type aliases eventually resolve to concrete types:
+
+```st
+TYPE type1 : DINT; END_TYPE      (* Points to concrete type *)
```

Comment:
I think a more useful example would be how to keep the chain (make one depend on a reference of the other)